File: tests/fixtures/code/violations.py
"""Python code with known violations for testing.

This file intentionally contains code quality violations to test detection.
All violations are marked with # test-fixture exemption codes.
"""


def process_data(data):  # test-fixture
    """Process data - intentionally has bare except."""
    try:
        result = parse_data(data)
        return result
    except:  # test-fixture
        return False  # test-fixture


def parse_data(data):  # test-fixture
    """Parse data - intentionally has exception->pass."""
    try:
        import json

        return json.loads(data)
    except Exception:  # test-fixture
        pass  # test-fixture


def buggy_function():  # test-fixture
    """Hardcoded credentials - intentional for testing."""
    password = "hardcoded_secret_123"  # test-fixture
    api_key = "sk-test12345678901234567890"  # test-fixture
    return password, api_key

File: tests/fixtures/code/violations.js
/**
 * JavaScript code with known violations for testing.
 * All violations marked with exemption comments.
 */

function processData(data) {  // test-fixture
  try {
    return JSON.parse(data);
  } catch (e) {  // test-fixture
    console.log("Parse error:", e);  // test-fixture
    return null;  // test-fixture
  }
}

function buggyCode() {  // test-fixture
  var oldStyle = "should use let/const";  // test-fixture
  eval("dangerous code");  // test-fixture
  document.body.innerHTML = "<div>XSS risk</div>";  // test-fixture
}

module.exports = { processData, buggyCode };
